index.bs

```diff
@@ -24,9 +46,66 @@ Please join us in the <a href="https://github.com/terinjokes/console-spec/issues
 <h2 id="logger-function">Logger Function</h2>
 The logging function accepts datum and the logLevel, which is stored or displayed in a platform appropriate way.
 
-TODO: Describe the interaction of format specifiers and object inspection with respect to what is stored or displayed
-
-TODO: Add notes to give examples of how a platform might implement the side effects of this logging function.
+Each call of the Logger function prints a line which is ended by a newline.
+If the Logger function is called with no arguments, no output is written to the console.
+If the printing function gets multiple elements, it prints them separated by a space. It is possible to pass format specifiers to the function as the element of the arguments that should get printed.
+
+The Logger(logLevel, multipleArgs) function must run these steps:
+
+<emu-alg>
+  1. Let _firstElement_ be the first element of _multipleArgs_.
+  1. Let _rest_ be all elements following _firstElement_ in _multipleArgs_.
+  1. Let _formatter_ be a recursive function.
+  1. If no _firstElement_ given, do not print any output. Abort these steps.
+  1. If just _firstElement_ given, print _firstElement_ to the console. Abort these steps.
+  1. If _firstElement_ does not contain any format specifier, print _multipleArgs_ separated by one space to the console. Abort these steps.
+  1. Call _formatter_ with _multipleArgs_:
+    1. Let _target_ be the first element of _multipleArgs_.
+    1. Let _current_ be the second element of _multipleArgs_.
+    1. Find the first possible format specifier from the left to the right in _target_.
+      1. If it is `%s`, let _converted_ be the result of new String(_current_).
+      1. If it is `%d` or `%i`, let _converted_ be the result of Number.parseInt(_current_, 10).
+      1. If it is `%f`, let _converted_ be the result of  Number.parseFloat(_current_, 10).
+      1. Replace _current_ in _target_ with _converted_.
+      1. Create _result_ which contains _target_ together with the elements of _multipleArgs_, but with _current_ removed.
+    1. If _target_ does not have any format specifiers left, return _result_ and end the recursive function. Print the result. Abort the steps.
+    1. If _result_ is just one element, return _result_ and end the recursive function. Print the result. Abort the steps.
+    1. Call _formatter_ with _result_ until it returns.
+</emu-alg>
```

Comment:
I think maybe I get it. What you want is to create a second abstract operation, Formatter(multipleArgs), which is defined in a separate section similar to Logger. That will be better than nesting the definition of the formatter function inside the definition of Logger.